### PR TITLE
Add PRISMA dataset

### PR DESCRIPTION
**Pull request by adamjstewart**
PRISMA data is distributed in HDF5 files. At the moment, TorchGeo doesn't know how to reproject/resample HDF5 files (pending #1490). Until then, I wrote a data loader that assumes a user has already manually converted all files from HDF5 to something that GDAL can read.

- [x] Data loading
- [x] Plotting
- [x] Testing
- [x] Documentation

prisma

### CODE REVIEW

File: docs/api/geo_datasets.csv [hunk 1]
```diff
@@ -18,6 +18,7 @@ Dataset,Type,Source,Size (px),Resolution (m)
 `Landsat`_,Imagery,Landsat,"8,900x8,900",30
 `NAIP`_,Imagery,Aerial,"6,100x7,600",1
 `NLCD`_,Masks,Landsat,-,30
-`NCCM`_,Masks,Sentinel-2,-,10 
+`NCCM`_,Masks,Sentinel-2,-,10
 `Open Buildings`_,Geometries,"Maxar, CNES/Airbus",-,-
+`PRISMA`_,Imagery,PRISMA,512x512,5--30
```
Comment (adamjstewart) [inline docs/api/geo_datasets.csv:23]: The files I was given are 512x512, need to see if that's the size of all files or just this subset.

**Comment by calebrob6** [comment]
Do you have a script for "manually convert all files from HDF5 to something that GDAL can read."?

**Comment by adamjstewart** [comment]
Yep:
```python
import glob
import os

import h5py
import numpy as np
import rasterio as rio


hdf5_dir = "hdf5"
gtif_dir = "gtif"

for file in glob.iglob(os.path.join(hdf5_dir, "*.he5")):
    print(file)
    with h5py.File(file) as src:
        epsg = src.attrs["Epsg_Code"]

        west = min(src.attrs["Product_LLcorner_easting"], src.attrs["Product_ULcorner_easting"])
        south = min(src.attrs["Product_LLcorner_northing"], src.attrs["Product_LRcorner_northing"])
        east = max(src.attrs["Product_LRcorner_easting"], src.attrs["Product_URcorner_easting"])
        north = max(src.attrs["Product_ULcorner_northing"], src.attrs["Product_URcorner_northing"])

        vnir = src["HDFEOS/SWATHS/PRS_L2D_HCO/Data Fields/VNIR_Cube"][()]
        swir = src["HDFEOS/SWATHS/PRS_L2D_HCO/Data Fields/SWIR_Cube"][()]

    data = np.concatenate([vnir, swir], axis=1)
    data = np.transpose(data, [1, 0, 2])
    count, height, width = data.shape

    profile = {
        "fp": file.replace(hdf5_dir, gtif_dir).replace(".he5", ".tif"),
        "mode": "w",
        "driver": "GTiff",
        "width": width,
        "height": height,
        "count": count,
        "crs": rio.crs.CRS.from_epsg(epsg),
        "transform": rio.transform.from_bounds(west, south, east, north, width, height),
        "dtype": data.dtype,
    }

    with rio.open(**profile) as dst:
        for i in range(count):
            dst.write(data[i], i + 1)
```

**Comment by calebrob6** [comment]
Maybe you can create a gist with that script and link to it from the docstring?

**Review by calebrob6 — DISMISSED** [review]

**Review by calebrob6 — APPROVED** [review]

**Pull request merged**